solc/CommandLineInterface.cpp

```diff
@@ -595,16 +589,7 @@ bool CommandLineInterface::compile()
 		m_compiler->enableIRGeneration(m_options.compiler.outputs.ir || m_options.compiler.outputs.irOptimized);
 		m_compiler->enableEwasmGeneration(m_options.compiler.outputs.ewasm);
 
-		OptimiserSettings settings = m_options.optimizer.enabled ? OptimiserSettings::standard() : OptimiserSettings::minimal();
-		if (m_options.optimizer.expectedExecutionsPerDeployment.has_value())
-			settings.expectedExecutionsPerDeployment = m_options.optimizer.expectedExecutionsPerDeployment.value();
-		if (m_options.optimizer.noOptimizeYul)
-			settings.runYulOptimiser = false;
-
-		if (m_options.optimizer.yulSteps.has_value())
-			settings.yulOptimiserSteps = m_options.optimizer.yulSteps.value();
-		settings.optimizeStackAllocation = settings.runYulOptimiser;
```

Comment:
Note that this line was not in `assemble()`, only in `compile()`. I removed it because it's redundant anyway. With current defaults in `OptimierSettings` the value of `optimizeStackAllocation` always matches `runYulOptimiser`. On the CLI they can only be different if you use `--no-optimize-yul` and that option is invalid in assembly mode.